src/cargo/core/resolver/context.rs

```diff
@@ -26,6 +26,12 @@ pub struct Context {
     pub activations: Activations,
     pub resolve_features: im_rc::HashMap<PackageId, Rc<HashSet<InternedString>>>,
     pub links: im_rc::HashMap<InternedString, PackageId>,
+    pub public_dependency:
+        im_rc::HashMap<PackageId, im_rc::HashMap<InternedString, (PackageId, bool)>>,
+
+    // This is somewhat redundant with the `resolve_graph` that stores the same data,
+    //   but for querying in the opposite order.
+    pub parents: Graph<PackageId, Rc<Vec<Dependency>>>,
```

Comment:
In our past debugging we've found that cloning a `Context` happens so often that an expensive clone means a slower resolver, so I was curious about this! 

I think we explicitly removed `Graph` from this data structure earlier in favor of `RcList` below, so was curious on your thoughts on this.

I haven't read the whole patch yet though, so I'll likely find what happens soon.